cmd/snap/cmd_quota.go

```diff
@@ -118,13 +171,61 @@ func parseQuotas(maxMemory string) (*client.QuotaValues, error) {
 		mem = value
 	}
 
+	if cpuMax != "" {
+		countValue, percentageValue, err := parseCpuQuota(cpuMax)
+		if err != nil {
+			return nil, err
+		}
+		if percentageValue > 100 || percentageValue <= 0 {
+			return nil, fmt.Errorf("cannot use value %v: cpu quota percentage must be between 1 and 100", percentageValue)
+		}
+
+		cpuCount = countValue
+		cpuPercentage = percentageValue
+	}
+
+	if cpuSet != "" {
+		cgv, err := cgroupVersion()
+		if err != nil {
+			return nil, err
+		}
+		if cgv < 2 {
+			return nil, fmt.Errorf("cannot use CPU set with cgroup version %d", cgv)
+		}
```

Comment:
Should we check cgroup version on the client? Cerainly no harm but we need to do this in snapd anyway right?

Reply:
That's a very good point! I had a look at the current code: we do have some service-side validation in `snap/quota/resources.go`, but a check on the cgroup version is missing.

Reply:
I did a followup https://github.com/snapcore/snapd/pull/11577 for this